packages/mobile/src/services/offline-downloader/workers/utils.ts

```diff
@@ -0,0 +1,14 @@
+import { reachabilitySelectors } from '@audius/common'
+import queue from 'react-native-job-queue'
+
+import { store } from 'app/store'
+
+const getIsReachable = reachabilitySelectors.getIsReachable
+
+export const startQueueIfOnline = async () => {
+  const state = store.getState()
+  const reachable = getIsReachable(state)
+  if (reachable) {
+    return queue.start()
```

Comment:
Because this can fail when we're not online, we could miss a window where we intended to start the queue but skipped it. I think we may need something to start the queue whenever we become reachable. This could also be a problem though, since some operations want to stop the queue temporarily. We might need some sort of mutex?

Reply:
@amendelsohn  we do have already an effect to start the queue whenever we become reachable 
https://github.com/AudiusProject/audius-client/blob/nkang--queue/packages/mobile/src/components/offline-downloads/OfflineDownloader.tsx#L41-L50

Reply:
Ah, you're right. And I'm slowly convincing myself that the race condition is invalid... not sure